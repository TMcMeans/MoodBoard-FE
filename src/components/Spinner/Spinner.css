.spinner-wrapper {
  width: 100wh;
  height: 100vh;
  background: white;
  margin: 0 auto 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.spinner {
  height: 200px;
  width: 200px;
  margin: 0 auto 0 auto;
}
