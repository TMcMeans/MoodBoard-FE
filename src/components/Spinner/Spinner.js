import React from 'react';
import spinner from '../../assets/spinner.gif';
import './Spinner.css';

export const Spinner = ({ text }) => {
  return (
    <div className="spinner-wrapper">
      <img src={spinner} alt="Loading icon" className="spinner" />
      <h1>{text}</h1>
    </div>
  );
};
