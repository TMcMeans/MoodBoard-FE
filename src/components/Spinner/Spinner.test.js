import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from './Spinner';

describe('Logo', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Spinner />);
  });
  it('should render a snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
