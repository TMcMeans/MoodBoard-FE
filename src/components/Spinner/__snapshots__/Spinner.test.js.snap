// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render a snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Loading icon"
          className="spinner"
          src="spinner.gif"
        />,
        <h1 />,
      ],
      "className": "spinner-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Loading icon",
          "className": "spinner",
          "src": "spinner.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Loading icon"
            className="spinner"
            src="spinner.gif"
          />,
          <h1 />,
        ],
        "className": "spinner-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Loading icon",
            "className": "spinner",
            "src": "spinner.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
